evm_arithmetization/src/fixed_recursive_verifier.rs

```diff
@@ -868,6 +960,85 @@ where
         }
     }
 
+    /// Create two-to-one block aggregation circuit.
+    ///
+    /// # Arguments
+    ///
+    /// - `block_circuit`: circuit data for the block circuit, that constitues
+    ///   the base case for aggregation.
+    ///
+    /// # Outputs
+    ///
+    /// Returns a [`TwoToOneBlockCircuitData<F, C, D>`].
+    fn create_two_to_one_block_circuit(
+        block_circuit: &BlockCircuitData<F, C, D>,
+    ) -> TwoToOneBlockCircuitData<F, C, D>
+    where
+        F: RichField + Extendable<D>,
+        C: GenericConfig<D, F = F>,
+        C::Hasher: AlgebraicHasher<F>,
+    {
+        let mut builder = CircuitBuilder::<F, D>::new(block_circuit.circuit.common.config.clone());
+
+        let mix_hash = builder.add_virtual_hash_public_input();
+
+        // We need to pad by PIS to match the count of PIS of the `base_proof`.
+        let mut padding = block_circuit.circuit.common.num_public_inputs;
+        // The number of PIS that will be added *after* padding by
+        // [`add_verifier_data_public_inputs()`].
+        padding -= verification_key_len(&block_circuit.circuit);
+        // Account for `mix_pv_hash`.
+        padding -= builder.num_public_inputs();
+
+        for _ in 0..padding {
```

Comment:
I _think_ you can do something like this?
```
let zero = builder.zero();
builder.register_public_inputs(&vec![zero; padding]);
```

Reply:
yes but this would allocate the vec.

Reply:
Hmm, true. Then as a nit you can put the zero target outside of the loop.